packages/styled-components/src/types.ts

```diff
@@ -243,12 +243,12 @@ export type CSSPseudos = { [K in CSS.Pseudos]?: CSSObject };
 
 export type CSSKeyframes = object & { [key: string]: CSSObject };
 
-export type CSSObject = StyledObject<any>;
+export type CSSObject<Props extends object = BaseObject> = StyledObject<Props>;
 
-export type StyledObject<Props extends object> = CSSProperties &
+export type StyledObject<Props extends object = BaseObject> = CSSProperties &
   CSSPseudos & {
     [key: string]:
-      | StyledObject<any> // StyledObject<any> instead of CSSObject. Because writing CSSObject directly results in a circularly references.
```

Comment:
Hey @takurinton, can we change this back to `StyleObject<Props>`?  I think it should work as long as we are referencing `StyleObject` and not `CSSObject`, right? The circular reference would only occur if we referenced `CSSObject` here.

Or Is there another reason that this causes a problem?

Reply:
That's right, I think it's good!!